product_code/course_service/impl/src/main/scala/de/upb/cs/uc4/course/impl/CourseServiceImpl.scala

```diff
@@ -140,7 +148,14 @@ class CourseServiceImpl(
           throw UC4Exception.PathParameterMismatch
         }
 
-        val validationErrors = updatedCourse.validate
+        val validationErrors = try {
+          Await.result(updatedCourse.validate, 5.seconds)
```

Comment:
The timeout should be defined in the application.conf
And than loaded into a lazy val